x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigureNode.java

```diff
@@ -383,7 +386,7 @@ protected void execute(Terminal terminal, OptionSet options, Environment env) th
                 transportKey = transportKeyPair.getPrivate();
                 transportCert = CertGenUtils.generateSignedCertificate(
                     certificatePrincipal,
-                    getSubjectAltNames(),
+                    null, // transport only validates certificates not the hostname
```

Comment:
good call. Not only these are not used for validation but will probably ( apart from the localhost related ones ) be _wrong_ for the other nodes that will get this certificate :+1: